### PR TITLE
Feature/hover media query

### PR DESCRIPTION
**Pull request by ReneDrie**
Uses the (hover: hover) media query so we don't need additional checks.
https://developer.mozilla.org/en-US/docs/Web/CSS/@media/hover

### CODE REVIEW

**Review by tworrisb — APPROVED** [review]

File: utils/mixin/_hover.scss [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
- * Leaving it to just :hover by default. But best practice to always use the mixin
+ * Uses the hover mixin with an optional extra selector (like '$:focus-visible')
  *
  * @param $extraSelector:string (optional) = an extra selector that can have the same styles as a hover (for mobile/active state)
  */
 @mixin hover($extraSelector: null) {
-  &:hover#{", "$extraSelector} {
-    @content;
-  }
+	@media (hover: hover) {
```
Comment (ThaNarie) [inline utils/mixin/_hover.scss:9]: This wil not work on IE11, right? I would suggest to:

* Either add this "workaround"; https://dev.to/niorad/detecting-hover-and-touch-in-css-and-js-4e42
* Or bump the major version to `2.0.0` as a breaking change, and document that IE11 is not supported anymore

**Review by ThaNarie — CHANGES_REQUESTED** [review]

File: utils/mixin/_hover.scss (hunk 1 above)
Comment (tworrisb) [reply utils/mixin/_hover.scss:9]: Your right, I would suggest adding the IE11 hack. I think we still get projects with IE11 support right?

Comment (psimk) [reply utils/mixin/_hover.scss:9]: @ThaNarie @tworrisb 

We have been in talks with @pigeonfresh to revitalize the whole `seng-scss` package as `@mediamonks/scss-utils`. A rename, means a new package, so we can afford adding breaking changes to it. Would both of you agree with merging this PR after we do the transition to `@mediamonks/scss-utils`?

The transition includes the completion of these two issues: 
1. https://github.com/mediamonks/seng-scss/issues/15
2. https://github.com/mediamonks/seng-scss/issues/16

Comment (ReneDrie) [reply utils/mixin/_hover.scss:9]: I would as well

**Review by ThaNarie — APPROVED** [review]

**Pull request merged**